### PR TITLE
Fix potential double-deletion of objects in ROOT

### PR DESCRIPTION
**Pull request by linev**
Found  by DeepCode analyzer

### CODE REVIEW

**Comment by linev** [comment]
@eguiraud Can be https://github.com/root-project/root/pull/7469/commits/cb41756658bf8046f41ec60f126e680fba19c4fc reason for last errors I saw in Jenkins?

**Comment by eguiraud** [comment]
@linev I don't know, what errors? :D

**Comment by linev** [comment]
@eguiraud Why I submit too many PRs, I see very often errors like here:
https://lcgapp-services.cern.ch/root-jenkins/job/root-pullrequests-build/110020/console
And `gtest-roofit-RDataFrameHelpers-test-testActionHelpers` appears very often

**Comment by eguiraud** [comment]
Uhm no, `gtest-roofit-RDataFrameHelpers-test-testActionHelpers` is unrelated to the changes in cb41756 (which I will review properly as soon as possible)

**Review by pcanal — APPROVED** [review]

**Comment by linev** [comment]
@eguiraud 
Can you approve too?
Only in RDF it is not only reset pointers to nullptr after deletion (which should be safe operation)
but also memory leak in ActionHelpers.

File: gui/sessionviewer/src/TProofProgressLog.cxx
Comment (pcanal) [inline gui/sessionviewer/src/TProofProgressLog.cxx:660]: As a side note, _historically_ we have used the ROOT macro `SafeDelete` for this pattern.

Comment (linev) [reply gui/sessionviewer/src/TProofProgressLog.cxx:660]: From other side, all kinds of code profilers do not understand them.
Much easier to write three lines and make them (clang-tidy or DeepCode) happy
All my last leaks I found via DeepCode

Comment (pcanal) [reply gui/sessionviewer/src/TProofProgressLog.cxx:660]: > All my last leaks I found via DeepCode

And thank you for fixing them ! :)

> From other side, all kinds of code profilers do not understand them.

That would be surprising, it literally expand to the same code (so only the code profilers that don't understand C macro will miss it ...)

**Comment by eguiraud** [comment]
@linev yes I'll review tomorrow, please wait until then to merge :)

**Review by eguiraud — APPROVED** [review]
thanks @linev , looks good to me

**Pull request merged**
